.github/scripts/test_record_expiry.py

```
@@ -0,0 +1,317 @@
+        self.bind(MAC, term="2027-01-01T00:00:00")
+        self.bind(WIN)
+
+        with self.assertRaises(SystemExit):
+            self.run_script(WIN)
+
+    def test_a_corrupt_sidecar_does_not_silently_renew(self):
+        """Treating unreadable state as absent would turn corruption into a renewal."""
+        self.bind(MAC)
+        (self.licenses / f"{MAC}.meta.json").write_text("{not json")
+
+        with self.assertRaises(Exception) as raised:
```

Ruff (0.16.3)

[warning] Use `pytest.raises` instead of unittest-style `assertRaises`

Replace `assertRaises` with `pytest.raises`

(PT027)